test/new-e2e/tests/agent-shared-components/forwarder/nss_failover_test.go

```diff
@@ -0,0 +1,302 @@
+	"github.com/DataDog/test-infra-definitions/scenarios/aws/vm/ec2vm"
+	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	fi "github.com/DataDog/datadog-agent/test/fakeintake/client"
+	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/utils/e2e"
+	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/utils/e2e/client"
+)
+
+type multiFakeIntakeEnv struct {
+	VM          client.VM
+	Agent       client.Agent
+	Fakeintake1 *client.Fakeintake
+	Fakeintake2 *client.Fakeintake
+}
+
+const (
+	// intakeName should only contains alphanumerical characters to ease pattern matching /etc/hosts
+	intakeName              = "ddintake"
+	connectionResetInterval = 120 // seconds
+	intakeMaxWaitTime       = 5 * time.Minute
+	intakeTick              = 20 * time.Second
+	fakeintake1Name         = "1fakeintake"
+	fakeintake2Name         = "2fakeintake"
+)
+
+//go:embed testfixtures/custom_logs.yaml
+var customLogsConfig string
+
+func multiFakeintakeStackDef(agentOptions ...agentparams.Option) *e2e.StackDefinition[multiFakeIntakeEnv] {
+	return e2e.EnvFactoryStackDef(func(ctx *pulumi.Context) (*multiFakeIntakeEnv, error) {
+		awsEnv, err := awsResources.NewEnvironment(ctx)
+		if err != nil {
+			return nil, err
+		}
+
+		vm, err := ec2vm.NewEC2VMWithEnv(awsEnv)
+		if err != nil {
+			return nil, err
+		}
+
+		fiExporter1, err := aws.NewEcsFakeintake(awsEnv, fakeintakeparams.WithName(fakeintake1Name), fakeintakeparams.WithoutLoadBalancer())
+		if err != nil {
+			return nil, err
+		}
+
+		fiExporter2, err := aws.NewEcsFakeintake(awsEnv, fakeintakeparams.WithName(fakeintake2Name), fakeintakeparams.WithoutLoadBalancer())
+		if err != nil {
+			return nil, err
+		}
+
+		agentInstaller, err := agent.NewInstaller(vm, agentOptions...)
+		if err != nil {
+			return nil, err
+		}
+
+		return &multiFakeIntakeEnv{
+			VM:          client.NewPulumiStackVM(vm),
+			Agent:       client.NewPulumiStackAgent(agentInstaller),
+			Fakeintake1: client.NewFakeintake(fiExporter1),
+			Fakeintake2: client.NewFakeintake(fiExporter2),
+		}, nil
+	})
+}
+
+type multiFakeIntakeSuite struct {
+	e2e.Suite[multiFakeIntakeEnv]
+}
+
+func TestMultiFakeintakeSuite(t *testing.T) {
+	e2e.Run(t, &multiFakeIntakeSuite{}, multiFakeintakeStackDef())
+}
+
+// SetupSuite waits for both fakeintakes to be ready before running tests.
+func (v *multiFakeIntakeSuite) SetupSuite() {
+	v.Env() // update the environment outside EventuallyWithT
+
+	// Wait for the fakeintake to be ready to avoid 503
+	require.EventuallyWithT(v.T(), func(c *assert.CollectT) {
+		assert.NoError(c, v.Env().Fakeintake1.Client.GetServerHealth())
+		assert.NoError(c, v.Env().Fakeintake2.Client.GetServerHealth())
+	}, intakeMaxWaitTime, intakeTick)
+}
+
+// BeforeTest flushes both fakeintakes before starting each test.
+func (v *multiFakeIntakeSuite) SetupTest() {
+	v.NoError(v.Env().Fakeintake1.FlushServerAndResetAggregators())
```

Comment:
It looks like to be the behaviour we want in most cases, right? 
Did you consider moving into `e2e.Test`  (I can help you done that). @pducolin what do you think?

Reply:
> It looks like to be the behaviour we want in most cases, right?

I agree, not having this seems like a strong source of potential flakes/errors
I'd rather implement that in a different PR though, this one is already big / old enough :smile:

Reply:
On the CI this should not be required, I see it as a requirements in dev mode only. Does the test fails on CI without this ?

We can add it to the fakeintake client, in dev mode only maybe ? But fakeintake does not know about devmode. Thinking out loud, the fakeintake client usually connects to an ephemeral fakeintake server, so no need to flush it. It is a requirement if we connect to a fakeintake that already exists: this happens when working in dev mode. 

I wonder if we would miss valid information by flushing a brand new fakeintake. Is there something we would miss flushing it ?